### DISCUSSION TITLE
Add semicolon at the end of a line

### DISCUSSION THREAD

**Discussion by cd-a** [helix-editor/helix#9118] [Q&A]
I currently use this to add commas and semicolons to the end of the line

```toml
[keys.normal.space]
# Add comma
"," = [
    "save_selection",
    "goto_line_end",
    ":append-output echo ','",
    "collapse_selection",
    "delete_selection",
    "jump_backward",
]
# Add semicolon
";" = [
    "save_selection",
    "goto_line_end",
    ":append-output echo ';'",
    "collapse_selection",
    "delete_selection",
    "jump_backward",
]
```

It works great, but the undo is a bit messed up. 

Is there a way to improve it so just pressing undo once, it will just remove the semicolon and (optionally) keep me at the same location?

**Comment by Zyhael** [comment]
This is the way I have configured mine.
Undo removes all the semicolons and gives you back your selections before doing the command.

```
[keys.normal]
"C-;" = [ 
    "goto_line_end", 
    ":append-output echo ;", 
    "collapse_selection", 
    "keep_primary_selection", 
]
```
